azext_iot/operations/events3/_events.py

```diff
@@ -127,12 +127,14 @@ def _get_conn_props():
     def _output_msg_kpi(msg):
         # TODO: Determine if amqp filters can support boolean operators for multiple conditions
         origin = str(msg.annotations.get(b'iothub-connection-device-id'), 'utf8')
-        if device_id:
-            regex = re.escape(device_id).replace("\\*", ".*").replace('\\?', ".") + "$"
-            if not re.match(regex, origin):
+        if device_id and device_id != origin:
+            if '*' in device_id or '?' in device_id:
```

Comment:
Consider putting * and ? in an array and then do lookup.

Reply:
In my testing, two string lookups with `in` performed much better than, say: 
`any(x in <string> for x in ['*', '?']`

`timeit.timeit("'*' in 'random_string' or '?' in 'random_string'")
0.07972277299995767
 timeit.timeit("any(x in 'random_string' for x in ['*', '?'])")
0.4291381340000271`

We could loop through a 'wildcard_chars' array and check for each wildcard character, but for only two characters is it necessary?

Reply:
excellent.  Thanks for doing the research on that.  It might warrant a blog post and further research to see at what point someone should consider an array lookup.